src/realm/object-store/property.hpp

```diff
@@ -336,6 +344,15 @@ inline Property::Property(std::string name, PropertyType type, std::string objec
 {
 }
 
+inline Property::Property(std::string name, const std::vector<PropertyType>& nested_types, std::string public_name)
```

Comment:
This is just the initial proposition. Essentially, we go super explicit. We keep a vector of `PropertyTypes` and we translate them to `CollectionTypes` when a new column is created.

Reply:
This seems reasonable.